tests/e2e/pageobjects/ide/PreviewWidget.ts

```diff
@@ -86,6 +86,8 @@ export class PreviewWidget {
 
         Logger.debug(`PreviewWidget.waitContentAvailable ${contentLocator}`);
 
+        await this.driverHelper.wait(polling);   // workaround https://github.com/eclipse/che/issues/19895
```

Comment:
Those hardcoded waits are not a good practice. I would instead try something from below solutions:

1. Set the environment variable `TS_SELENIUM_PREVIEW_WIDGET_DEFAULT_TIMEOUT`  (which is used directly in the `waitAndSwitchTOWidgetFrame` [method](https://github.com/eclipse/che/blob/384f90b6f07d7c3475ca9239f4f2b365e68b536c/tests/e2e/pageobjects/ide/PreviewWidget.ts#L73)) to something larger than default (10_000 ms) so it will be increased for your particular test.
2. Increase the default value of `TS_SELENIUM_PREVIEW_WIDGET_DEFAULT_TIMEOUT` [here](https://github.com/eclipse/che/blob/384f90b6f07d7c3475ca9239f4f2b365e68b536c/tests/e2e/TimeoutConstants.ts#L173), so it will be larger in every test using that method.
3. Leave the default timeout as is and add an optional timeout parameter to the `waitAndSwitchToWidgetFrame` method so the timeout can be adjusted per method call. 

If you need any help solving that issue, feel free to ping me.

Reply:
If you trying to fix a problem with a blank screen, unfortunately, a simple wait will not help.
You need to gain an application route, and if a blank screen shown type this route to the widget address bar.

Another way is to open the application in the browser tab instead preview widget.

Reply:
Example for using browser tab https://github.com/eclipse/che/blob/main/tests/e2e/tests/plugins/TypescriptPlugin.spec.ts#L98-L155

Reply:
Interesting, just for curiosity - is it known what is the root cause of this `blank screen` problem? Is there some issue created? Thanks

Reply:
As far as I remember it was some issue about that. But I am not sure. The main problem, this problem doesn't reproduce by hands. It is something specific for automation.

Reply:
@Katka92, @Ohrimenko1988: thank you for review!

@Katka92: The problem is that after switching to Preview Widget frame the test can't find "**Welcome**" message and [then instantly clicking on "refresh" button in Preview Widget, which redirects to "blank:empty" page](https://main-jenkins-csb-crwqe.apps.ocp4.prod.psi.redhat.com/job/Che/job/e2e/job/minikube/job/pr-check/job/che-theia-pr-check/137/consoleFull). 
There is an issue about broken refresh button exists https://github.com/eclipse/che/issues/19910
 
@Ohrimenko1988: actually, it has been working with simple 10 seconds wait.
E.g., you can compare:
- failed Happy path tests https://main-jenkins-csb-crwqe.apps.ocp4.prod.psi.redhat.com/job/Che/job/e2e/job/minikube/job/pr-check/job/che-theia-pr-check/137/consoleFull
- passed Happy path tests with timeout in the same PR: https://main-jenkins-csb-crwqe.apps.ocp4.prod.psi.redhat.com/job/Che/job/e2e/job/minikube/job/pr-check/job/che-theia-pr-check/143/consoleFull

IMHO, another possible "timeout less" solution in this case could be not clicking on "**Refresh**" button at all (`this.refreshPage()`), but waiting on page to be fully loaded in Preview Widget.

Reply:
I've searched a little in our codebase and there is already method including this polling/attmepts mechanism: https://github.com/eclipse/che/blob/3df3228210d625b490cf16ff2a926369d6f4ec14/tests/e2e/utils/DriverHelper.ts#L63 So maybe whole this [block](https://github.com/eclipse/che/pull/19909/files#diff-e3e126311533b5946de6dbfa699df2efaa80613f43750e687fa6a377f1aee855L90-L102) can be just replaced by the method I shared. But I'm not insisting on that - if you are OK with just removing the refresh part, I'm OK.

Reply:
And thank you @dmytro-ndp for sharing the link to the issue! :slightly_smiling_face: